drivers/storage/port/scsiport/ioctl.c

```diff
@@ -538,6 +538,11 @@ ScsiPortDeviceControl(
             status = STATUS_NOT_IMPLEMENTED;
             break;
 
+        case IOCTL_ATA_PASS_THROUGH:
+            DPRINT1("IOCTL_ATA_PASS_THROUGH is not supported. (Caller should use IOCTL_SCSI_PASS_THROUGH instead.)\n");
+            status = STATUS_NOT_IMPLEMENTED;
+            break;
```

Comment:
This would be better, also I believe MS scsiport returns different NTSTATUS here (need to double check)
```suggestion
        case IOCTL_ATA_PASS_THROUGH:
        case IOCTL_ATA_PASS_THROUGH_DIRECT:
            DPRINT1("Invalid IOCTL requested: 0x%lX, must be a bug on caller side.\n", Stack->Parameters.DeviceIoControl.IoControlCode);
            status = STATUS_NOT_IMPLEMENTED;
            break;
```

Reply:
The fact that the driver is unable to handle ATA passthrough IOCTLs doesn't necessarily mean that the caller is buggy.

Reply:
Done.